diff --git a/drivers/lora/CMakeLists.txt b/drivers/lora/CMakeLists.txt
index 9cd035fcf2b..cda2f6dbaf3 100644
--- a/drivers/lora/CMakeLists.txt
+++ b/drivers/lora/CMakeLists.txt
@@ -1,7 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
+# ZephCore patch: adds lr11xx driver subdirectory.
 
 zephyr_sources_ifdef(CONFIG_LORA_SHELL shell.c)
 
+add_subdirectory_ifdef(CONFIG_LORA_LR11XX lr11xx)
+
 # zephyr-keep-sorted-start
 add_subdirectory_ifdef(CONFIG_LORA_MODULE_BACKEND_LORAMAC_NODE loramac-node)
 add_subdirectory_ifdef(CONFIG_LORA_MODULE_BACKEND_LORA_BASICS_MODEM lora-basics-modem)
diff --git a/drivers/lora/Kconfig b/drivers/lora/Kconfig
index 657bdb9ce28..0ebee4246f3 100644
--- a/drivers/lora/Kconfig
+++ b/drivers/lora/Kconfig
@@ -5,6 +5,7 @@
 #
 
 # Top-level configuration file for LORA drivers.
+# ZephCore patch: adds lr11xx driver Kconfig.
 
 menuconfig LORA
 	bool "LoRa drivers"
@@ -60,6 +61,7 @@ config LORA_INIT_PRIORITY
 
 rsource "Kconfig.sx12xx"
 rsource "Kconfig.rylrxxx"
+rsource "lr11xx/Kconfig"
 rsource "lora-basics-modem/Kconfig"
 rsource "native/Kconfig"
 
